components/knowledge/OneDrive.tsx

```diff
@@ -135,6 +130,18 @@ export const OnedriveFileModal = ({
     }
   };
 
+  const clear = async () => {
+    setIsClearing(true);
+    try {
+      await clearOneDriveFiles();
+      syncOnedrive();
+    } catch (e) {
+      setSyncError((e as Error).toString());
+    } finally {
+      setIsClearing(false);
+    }
+  };
+
```

Comment:
nit: you can get most of this logic out of the box with the `useAsync` hook that I created